src/rules/variableNameRule.ts

```diff
@@ -60,130 +62,136 @@ export class Rule extends Lint.Rules.AbstractRule {
-            const identifier = node.name as ts.Identifier;
-            this.handleVariableNameFormat(identifier, node.initializer);
-            this.handleVariableNameKeyword(identifier);
+    function handleVariableNameFormat(name: ts.Identifier, initializer?: ts.Expression): void {
+        if (!options.checkFormat) {
+            return;
         }
-        super.visitVariableDeclaration(node);
-    }
 
-    public visitVariableStatement(node: ts.VariableStatement) {
-        // skip 'declare' keywords
-        if (!Lint.hasModifier(node.modifiers, ts.SyntaxKind.DeclareKeyword)) {
-            super.visitVariableStatement(node);
+        const { text } = name;
+        if (initializer && isAlias(text, initializer)) {
+            return;
         }
-    }
 
-    private isAlias(name: ts.Identifier, initializer: ts.Expression): boolean {
-        if (initializer.kind === ts.SyntaxKind.PropertyAccessExpression) {
-            return (initializer as ts.PropertyAccessExpression).name.text === name.text;
-        } else if (initializer.kind === ts.SyntaxKind.Identifier) {
-            return (initializer as ts.Identifier).text === name.text;
+        if (!isCamelCase(text, options) && !isUpperCase(text)) {
+            ctx.addFailureAtNode(name, Rule.FORMAT_FAILURE);
         }
-        return false;
     }
 
-    private handleVariableNameFormat(name: ts.Identifier, initializer?: ts.Expression) {
-        const variableName = name.text;
-
-        if (initializer && this.isAlias(name, initializer)) {
-            return;
-        }
-
-        if (this.shouldCheckFormat && !this.isCamelCase(variableName) && !isUpperCase(variableName)) {
-            this.addFailureAtNode(name, Rule.FORMAT_FAILURE);
+    function handleVariableNameKeyword(name: ts.Identifier): void {
+        if (options.banKeywords && BANNED_KEYWORDS.has(name.text)) {
+            ctx.addFailureAtNode(name, Rule.KEYWORD_FAILURE);
         }
     }
+}
 
-    private handleVariableNameKeyword(name: ts.Identifier) {
-        const variableName = name.text;
-
-        if (this.shouldBanKeywords && BANNED_KEYWORDS.indexOf(variableName) !== -1) {
-            this.addFailureAtNode(name, Rule.KEYWORD_FAILURE);
-        }
+function isAlias(name: string, initializer: ts.Expression): boolean {
+    switch (initializer.kind) {
+        case ts.SyntaxKind.PropertyAccessExpression:
+            return (initializer as ts.PropertyAccessExpression).name.text === name;
+        case ts.SyntaxKind.Identifier:
+            return (initializer as ts.Identifier).text === name;
+        default:
+            return false;
     }
+}
 
-    private isCamelCase(name: string) {
-        const firstCharacter = name.charAt(0);
-        const lastCharacter = name.charAt(name.length - 1);
-        const middle = name.substr(1, name.length - 2);
+function isCamelCase(name: string, options: Options): boolean {
+    const firstCharacter = name.charAt(0);
```

Comment:
`str.chatAt(pos)` vs `str[pos]` is your personal preference?
I prefer the latter, because it is shorter. But that's totally nice to have.